scipy/stats/_discrete_distns.py

```diff
@@ -653,6 +654,36 @@ def _argcheck(self, M, n, r):
         cond = (n >= 0) & (n <= M) & (r >= 0) & (r <= M-n)
         return cond
 
+    def _rvs(self, M, n, r, size=None, random_state=None):
+
+        def rvs1(M, n, r, size, random_state):
+            # invert cdf by calculating all values in support, scalar M, n, r
+            a, b = self.support(M, n, r)
+            ks = np.arange(a, b+1)
+            cdf = self.cdf(ks, M, n, r)
+            ppf = interp1d(cdf, ks, kind='next', fill_value='extrapolate')
+            rvs = ppf(random_state.random(size)).astype(int)
+            if size is None:
+                return rvs.item()
+            return rvs
+
+        M, n, r = np.broadcast_arrays(M, n, r)
+
+        if M.ndim == 0:
+            return rvs1(M, n, r, size, random_state)
+
+        res = np.zeros(size)
+
+        # move shape parameters axes to beginning for easier indexing
```

Comment:
The shape of the output is (shape of RVS for a given set of parameters) + (shape of broadcasted parameter arrays).

I iterate over the parameters with `np.ndenumerate`, which gives an index `i`. I want to use `i`  to index the part of the output array corresponding with one set of shape parameter.

I don't know how to do this when the output is in its original shape, so  I use `moveaxis` to change the shape of the output to (shape of broadcasted parameters) + (shape of RVS size for each set of parameters).

If anyone knows of a simpler way of using the indices `i` without having to move the axes around, I'd love to know about it!

Reply:
Factored this part out as a decorator in gh-13330.

Reply:
The broadcasting behavior is not quite right here, but I think it's fixed in gh-13330. Marking as draft until we can replace this with that decorator.